src/DynamoCoreWpf/ViewModels/Core/InPortViewModel.cs

```diff
@@ -0,0 +1,306 @@
+        /// </summary>
+        public Visibility UseLevelSpinnerVisible
+        {
+            get
+            {
+                if (PortType == PortType.Output) return Visibility.Collapsed;
+                if (UseLevels) return Visibility.Visible;
+                return Visibility.Hidden;
+            }
+        }
+
+        /// <summary>
+        /// If should keep list structure on this port.
+        /// </summary>
+        public bool ShouldKeepListStructure
+        {
+            get { return _port.KeepListStructure; }
+        }
+
+        /// <summary>
+        /// Levle of list.
+        /// </summary>
+        public int Level
+        {
+            get { return _port.Level; }
+            set
+            {
+                ChangeLevel(value);
+            }
+        }
+
+        /// <summary>
+        /// The visibility of Use Levels menu.
+        /// </summary>
+        public Visibility UseLevelVisibility
+        {
+            get
+            {
+                if (_node.ArgumentLacing != LacingStrategy.Disabled)
+                {
+                    return Visibility.Visible;
+                }
+                else
+                {
+                    return Visibility.Collapsed;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Sets the color of the small rectangular marker on each input port.
+        /// </summary>
+        public SolidColorBrush PortValueMarkerColor
+        {
+            get => portValueMarkerColor;
+            set
+            {
+                portValueMarkerColor = value;
+                RaisePropertyChanged(nameof(PortValueMarkerColor));
+            }
+        }
+
+        /// <summary>
+        /// Sets the color of the use levels popup in the input port context menu.
+        /// This changes when the Keep List Structure option is activated and the port
+        /// is connected, upon which it turns blue.
+        /// </summary>
+        public SolidColorBrush UseLevelsMenuColor
+        {
+            get
+            {
+                return ShouldKeepListStructure && _port.IsConnected
+                    ? new SolidColorBrush(Color.FromArgb(255, 60, 60, 60))
+                    : new SolidColorBrush(Color.FromArgb(255, 83, 83, 83));
+            }
+        }
+
+        #endregion
+
+        public InPortViewModel(NodeViewModel node, PortModel port) : base(node, port)
+        {
+            _port.PropertyChanged += _port_PropertyChanged;
+        }
+
+        public override void Dispose()
+        {
+            _port.PropertyChanged -= _port_PropertyChanged;
+            base.Dispose();
+        }
+
+        internal override PortViewModel CreateProxyPortViewModel(PortModel portModel)
+        {
+            return new InPortViewModel(_node, portModel);
+        }
+
+        void _port_PropertyChanged(object sender, System.ComponentModel.PropertyChangedEventArgs e)
```

Comment:
agree with @aparajit-pratap all these `_` are distracting given the move away from them in ours (and MS) conventions.